pandas/core/indexes/numeric.py

```diff
@@ -180,7 +180,8 @@ def _union(self, other, sort):
         if needs_cast:
             first = self.astype("float")
             second = other.astype("float")
-            return first._union(second, sort)
+            result = first._union(second, sort)
+            return Float64Index(result)
```

Comment:
this should already be a Float64Index.   are there cases where it isnt?

Reply:
Ah, you're right. At some point I was allowing `_union` to return just an array, so I ended up needing that. But now there's a `_shallow_copy` at return, so it's redundant. Will revert.